src/outlook.py

```diff
@@ -35,6 +38,51 @@ def getpolycenter(poly):
         round(sum(y for x, y in poly) / polylength, 2)
     )
 
+def getcentroids(outlook):
+    producttime = outlook['producttime']
+    day = datetime.strptime(outlook['spcdate'],'%Y%m%d')
+    path = day.strftime('data/%Y/%m/%d/shapefiles')
+    for wxtype in ('tornado', 'wind', 'hail'):
+        if wxtype == 'tornado': shp_suffix = '_torn.shp'
+        if wxtype == 'wind': shp_suffix = '_wind.shp'
+        if wxtype == 'hail': shp_suffix = '_hail.shp'
+        if wxtype == 'categorical': shp_suffix = '_cat.shp'
+        shapefile_path = path + '/day1otlk_' + outlook['spcdate'] + '_' + producttime + shp_suffix
+        if os.path.exists(shapefile_path):
+            shapefile = gp.read_file(shapefile_path)
+            for prob in ('0.60', '0.45', '0.30', '0.15', '0.10', '0.05', '0.02', 'SIGN'):
+                if prob in outlook['probabilistic'][wxtype]:
+                    if not prob == 'SIGN':
+                        intprob = int(prob[-2:])
+                    data = shapefile[shapefile['DN'] == intprob].to_crs(epsg=4326) # WGS 84
+                    if len(data) > 0:
+                        # TODO this currently only stores the last if there's multiple area polygons
+                        for index, row in data.iterrows():
+                            outlook['probabilistic'][wxtype][prob]['centroid'] = row['geometry'].centroid.coords[0]
+            del shapefile
+    
+    shapefile_path = path + '/day1otlk_' + outlook['spcdate'] + '_' + producttime + '_cat.shp'
+    if os.path.exists(shapefile_path):
+        shapefile = gp.read_file(shapefile_path)
+        cat_map = {
```

Comment:
This should be an enum and declared elsewhere. As it exists now, you're redeclaring it every time this runs.